### PR TITLE
add UniquePath

### PR DESCRIPTION
**Pull request by zzq1zh**
### **Describe your change:**

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.

### CODE REVIEW

File: src/main/java/com/thealgorithms/dynamicprogramming/UniquePaths.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        int n = 7;
+        System.out.println(uniquePaths2(3,7)); // result 28
+        System.out.println(uniquePaths2(3,2)); // result 3
+        System.out.println(uniquePaths2(3,3)); // result 6
+
+        System.out.println(uniquePaths(3,7)); // result 28
+        System.out.println(uniquePaths(3,2)); // result 3
+        System.out.println(uniquePaths(3,3)); // result 6
+    }
+
```
Comment (siriak) [inline src/main/java/com/thealgorithms/dynamicprogramming/UniquePaths.java:22]: Please remove main() and add some tests as in https://github.com/TheAlgorithms/Java/pull/2871/files

**Review by siriak — CHANGES_REQUESTED** [review]

**Comment by siddhant2002** [comment]
Can I work under this issue?

**Comment by zhrgithub** [comment]
Of  curse


| |
张泓锐
|
|
***@***.***
|
签名由网易邮箱大师定制


在2022年01月19日 12:39，Siddhant Swarup ***@***.***> 写道：

Can I work under this issue?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by siddhant2002** [comment]
@siriak please check the updated pr as I've added the tests.

**Pull request closed without merging**